### PR TITLE
fix(oci): oci downloader unavailable missing WithBundleParserOpts method

### PR DESCRIPTION
**Pull request by slonka**
### Why the changes in this PR are needed?

In https://github.com/open-policy-agent/opa/pull/6159 I introduced a disabled OCI downloader to get rid of dependency on containerd to minimise the CVE surface. https://github.com/open-policy-agent/opa/pull/6521/files#diff-f808512360f39e5c4f67342ca24d84117be5403637b032d67e82df3a425cdde9R139 broke our builds adding a method only to one of the implementations. 

### What are the changes in this PR?

In this PR I added an interface (should show up in IDE) and a CI check (makes sure building `make GO_TAGS="-tags=opa_no_oci" build` works) to see if everything builds with `GO_TAGS="-tags=opa_no_oci"`.

### Notes to assist PR review:

There supposed to be a long term solution for (getting rid of containerd) this but it seems that it was not implemented.

### Further comments:

Hope you don't mind me doing a PR - if you want to move this to an issue and discuss - feel free to do so, or tell me to do it :)

### CODE REVIEW

**Review by ashutosh-narkar — COMMENTED** [review]
Thanks for working on this @slonka! Instead of introducing an interface if we just implemented `WithBundleParserOpts` in the no OCI downloader this should work, correct? If yes, we can just do that and look into adding a new interface if needed in the future. WDYT?

Also can you please look into the test failures. Thanks.

File: .github/workflows/pull-request.yaml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
             capabilities.json
 
   go-build:
-    name: Go Build (${{ matrix.os }}${{ matrix.arch && format(' {0}', matrix.arch) || '' }})
+    name: Go Build (${{ matrix.os }}${{ matrix.arch && format(' {0}', matrix.arch) || '' }}${{ matrix.go_tags }})
```
Comment (ashutosh-narkar) [inline .github/workflows/pull-request.yaml:33]: What happens in the case `go_tags` is not set?

File: download/oci_download_unavailable.go [hunk 2]
```diff
@@ -51,3 +55,7 @@ func (d *OCIDownloader) Start(context.Context) {
 func (d *OCIDownloader) Stop(context.Context) {
 	panic("built without OCI support")
 }
+
+func (d *OCIDownloader) WithBundleParserOpts(ast.ParserOptions) *OCIDownloader {
```
Comment (srenatus) [inline download/oci_download_unavailable.go:59]: [nit]
```suggestion
func (*OCIDownloader) WithBundleParserOpts(ast.ParserOptions) *OCIDownloader {
```

File: .github/workflows/pull-request.yaml (hunk 1 above)
Comment (srenatus) [reply .github/workflows/pull-request.yaml:33]: It'll end up as nothing, I think. The CI run of this PR shows it. The only problem is that we seem to be unsetting some args what we need, too: https://github.com/open-policy-agent/opa/actions/runs/7802363034/job/21279888203

**Comment by slonka** [comment]
> Instead of introducing an interface if we just implemented WithBundleParserOpts in the no OCI downloader this should work, correct? If yes, we can just do that and look into adding a new interface if needed in the future. WDYT?

Yeah, it's not strictly needed but it might show up in some IDEs that a method is not implemented on this struct. A definite guard is what I've added in CI. I'll remove and we can add it in the future.

File: .github/workflows/pull-request.yaml (hunk 1 above)
Comment (slonka) [reply .github/workflows/pull-request.yaml:33]: Yes, it will just be empty. Adjusted the call so that it won't interfere with other tags.

**Review by ashutosh-narkar — APPROVED** [review]
LGTM. Can you please squash your commits and we can get this in.

File: .github/workflows/pull-request.yaml (hunk 1 above)
Comment (ashutosh-narkar) [reply .github/workflows/pull-request.yaml:33]: @srenatus does this address your concern?

**Comment by slonka** [comment]
@ashutosh-narkar squashed, can you re-approve so I can merge?

File: .github/workflows/pull-request.yaml (hunk 1 above)
Comment (srenatus) [reply .github/workflows/pull-request.yaml:33]: Tests pass, so I'm happy 😃 Thanks!

**Comment by slonka** [comment]
@ashutosh-narkar do you think we could get this released soon-ish?

**Comment by ashutosh-narkar** [comment]
@slonka we have a monthly release cadence. We'll have an OPA release end of this month.

**Pull request merged**